ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/validation/BlockValidator.java

```diff
@@ -94,6 +96,16 @@ public SafeFuture<InternalValidationResult> validate(SignedBeaconBlock block) {
       return completedFuture(reject("Parent block is after child block."));
     }
 
+    if (blobsSidecar.isPresent()) {
+      if (blobsSidecar.get().getBeaconBlockRoot().equals(block.getRoot())) {
```

Comment:
I don't understand this. Shouldn't it be opposite? And the next one

Reply:
Oops, I forgot a `!`!

Reply:
and btw both weights for reject from my side. 
I understand this one 
> [IGNORE] the sidecar.beacon_block_slot is for the current slot (with a MAXIMUM_GOSSIP_CLOCK_DISPARITY allowance) -- i.e. sidecar.beacon_block_slot == block.slot.
as slots for sidecar and block matches but it arrives too later or earlier. Where are your rules in spec?

Reply:
the `MAXIMUM_GOSSIP_CLOCK_DISPARITY` check is already implemented for block, while for the other checks I started a conversation here https://github.com/ethereum/consensus-specs/issues/3109

Reply:
moving validation to a separate PR